contracts/Removal.sol

```diff
@@ -442,6 +443,9 @@ contract Removal is
   ) internal override whenNotPaused {
     uint256 numberOfTokenTransfers = amounts.length;
     for (uint256 i = 0; i < numberOfTokenTransfers; ++i) {
+      if (amounts[i] == 0) {
```

Comment:
originally I had a different PR open for this and I'd put this inside `safeBatchTransferFrom` and Jaycen suggested moving it into `beforeTokenTransfer`.  I also thought about adding a check to see if it's being transferred to the market contract, and to only revert if that's the case.

@jaycenhorton  what do you think?
Personally I can't think of any scenarios where we would want to be able to allow removal balances of 0 to be transferred but I think there may still be a lot of unknown.  we can always update this logic in the future too

Reply:
The underlying 1155 implementation does **not** block transfers of an amount 0. Transfers of amount zero are technically valid according to the 1155 spec. That's one of the reasons I suggested adding it here instead. I think this check should stay here, with the primary justification being that if we did upgrade functionality in the future we wouldn't have to remember this.

From the spec: `To broadcast the existence of a token ID with no initial balance, the contract SHOULD emit the TransferSingle event from 0x0 to 0x0, with the token creator as _operator, and a _value of 0.`. And because it's in the spec, OZ is not preventing it.

I couldn't think of a single use case for such a thing for us since we only ever mint/burn/transfer removals of a value > 0. As such, I think it's best to prevent it in all possible cases.